framework/include/cppmicroservices/ServiceRegistrationBase.h

```diff
@@ -196,11 +196,11 @@ namespace cppmicroservices
          */
         ServiceRegistrationBase();
 
-        ServiceRegistrationBase(ServiceRegistrationBasePrivate* registrationPrivate);
+        ServiceRegistrationBase(std::shared_ptr<ServiceRegistrationBasePrivate> registrationPrivate);
 
         ServiceRegistrationBase(BundlePrivate* bundle, InterfaceMapConstPtr const& service, Properties&& props);
 
-        ServiceRegistrationBasePrivate* d { nullptr };
+        std::shared_ptr<ServiceRegistrationBasePrivate> d { nullptr };
```

Comment:
I think the nullptr is unecessary. {} will do the same thing

Reply:
Yep, I have changed it to just "std::shared_ptr<ServiceRegistrationBasePrivate? d;" in "Updated based on Patty's comments"(https://github.com/CppMicroServices/CppMicroServices/pull/841/commits/7d7bdd60e332b54a5954fe8823ccc2212b5a46d3)